lib/internal/freelist.js

```diff
@@ -1,24 +1,25 @@
 'use strict';
 
-// This is a free list to avoid creating so many of the same object.
-exports.FreeList = function(name, max, constructor) {
-  this.name = name;
-  this.constructor = constructor;
-  this.max = max;
-  this.list = [];
-};
-
-
-exports.FreeList.prototype.alloc = function() {
-  return this.list.length ? this.list.pop() :
-                            this.constructor.apply(this, arguments);
-};
+class FreeList {
+  constructor(name, max, constructor) {
+    this.name = name;
+    this.constructor = constructor;
+    this.max = max;
+    this.list = [];
+  }
 
+  alloc() {
+    return this.list.length ? this.list.pop() :
+                              this.constructor.apply(this, arguments);
```

Comment:
Unrelated to this PR, but perhaps this can be simplified since `http` is the only user of FreeList and we'd know how what arguments to pass on to the constructor. I'm not sure if adding `new` would help (allowing us to avoid an `instanceof` check in the constructor if possible) more than it might hurt (e.g. if stack traces would change with `new constructor(..)`).